elasticsearch/docker/templates/Dockerfile.j2

```diff
@@ -129,6 +169,9 @@ RUN mkdir /usr/share/supervisor
 
 WORKDIR /usr/share/elasticsearch
 COPY --from=prep_es_files --chown=1000:0 /usr/share/elasticsearch /usr/share/elasticsearch
+COPY --from=prep_es_files /usr/share/elasticsearch/k-NN/buildSrc/libKNNIndexV1_7_3_6.so /usr/lib
+
+RUN rm -rf /usr/share/elasticsearch/k-NN
```

Comment:
Why are we adding this layer to remove KNN ? Why it is required.

Reply:
We need to checkout KNN package to read the jni files for building the shared library. Once the shared library is build, we can remove the   repo

Reply:
I believe this can stay as is.
Please confirm

Reply:
Yes you are right. This needs to stay.

Reply:
@rishabh6788  If we need this could you combine it into with some other layer if possible.  This will create a new layer on docker which will impact the size If possible.

Reply:
We can also combine [this](https://github.com/opendistro-for-elasticsearch/opendistro-build/pull/74/files#diff-18fb87a79400756569b617b53c700eefR164) line to the above layer command.

Reply:
> We need to checkout KNN package to read the jni files for building the shared library. Once the shared library is build, we can remove the repo

Trying to understand more, if this can be deleted and it was built in prep stage, we may not need it here right ?

Reply:
Issue is due to https://github.com/opendistro-for-elasticsearch/opendistro-build/blob/knn-dockerfile/elasticsearch/docker/templates/Dockerfile.j2#L171 all the contents including kNN repo are getting copied.

Reply:
I am still confused, I know the size is almost same so not big concern, but we should investigate it why if you're just coping one file why it is brining back all the files?

Reply:
@mihirsoni  Pointed out some changes to move this to prep stage. We should be good after this

Reply:
Moved the repo deletion to prep stage and copying C lib file from tmp dir